### PR TITLE
[openshift-saas-deploy-trigger-cleaner] introduce new integration

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-3340

this PR introduces a new integration called `openshift-saas-deploy-trigger-cleaner`.
this integration goes over all pipelines providers, and for each one of type `tekton` it performs a cleanup according to an optinal `retention` section.

depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/20431

### CODE REVIEW

File: reconcile/openshift_saas_deploy_trigger_cleaner.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        integration=QONTRACT_INTEGRATION,
+        settings=settings, internal=internal,
+        use_jump_host=use_jump_host,
+        thread_pool_size=thread_pool_size)
+    defer(lambda: oc_map.cleanup())
+
+    for pp in pipelines_providers:
+        retention = pp.get('retention')
+        if not retention:
+            continue
```
Comment (BumbleFeng) [inline reconcile/openshift_saas_deploy_trigger_cleaner.py:45]: Should we hava default retention setting instead of ignoring those pipelines?

Comment (maorfr) [reply reconcile/openshift_saas_deploy_trigger_cleaner.py:45]: good next iteration? for now we have very few namespaces that have A LOT of PipelineRuns.

Comment (rrati) [reply reconcile/openshift_saas_deploy_trigger_cleaner.py:45]: Next iteration wfm

Comment (jmelis) [reply reconcile/openshift_saas_deploy_trigger_cleaner.py:45]: +1

**Pull request merged**